packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -189,13 +205,29 @@ class Hero extends StatefulWidget {
   ///
   /// If none is provided, the destination route's Hero child is shown in-flight
   /// by default.
+  ///
+  /// ## Limitations
+  ///
+  /// If a widget built by [flightShuttleBuilder] takes part in a [Navigator]
+  /// push transition, that widget or its descendants must not have any
+  /// [GlobalKey] that is used in the source Hero's descendant widgets, as both
```

Comment:
Better to break the "as both" point into a new sentence:

...widgets. Both subtrees will be included ... animation, and global keys must be unique across the entire widget tree.